rc/tools/patch.kak

```diff
@@ -0,0 +1,160 @@
+define-command patch -params .. -docstring %{
+    patch [<arguments>]: apply selections in diff to a file
+
+    Given some selections within a unified diff, apply the changed lines in
+    each selection by piping them to "patch <arguments> 1>&2"
+    (or "<arguments> 1>&2" if <arguments> starts with a non-option argument).
```

Comment:
What may be confusing is that `patch some-file.diff` does not work.
We could try to make it work -- for example by using selections only if either no non-option argument is present or there is a "-" argument.
(Also that would mean we couldn't use the proposed `<arguments>` trick that enables "patch git apply".
That wouldn't be a big deal, we could pass "git apply" via an option or register..)

However there is not much point to `:patch some-file.diff` when we can do `<a-!>patch some-file.diff` so I don't think we care